src/main/java/seedu/address/model/ingredient/IngredientQuantity.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.model.ingredient;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * A class to represent the ingredientQuantity in an ingredient
+ */
+
+public class IngredientQuantity {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Ingredient's quantity should be non-zero unsigned integer.";
+
+    // Identity fields
+    private int ingredientQuantity;
+    /**
+     * Constructs a {@code IngredientQuantity}.
+     *
+     * @param quantity A valid quantity, an integer that is non-negative.
+     */
+    public IngredientQuantity(int quantity) {
+        this.ingredientQuantity = quantity;
+    }
+
+    public int getQuantity() {
+        return ingredientQuantity;
+    }
+
+    /**
+     * Returns true if ingredientQuantity is valid.
+     * @param test
+     * @return
+     */
+    public static boolean isValidIngredientQuantity(String test) {
+        if (StringUtil.isNonZeroUnsignedInteger(test)) {
```

Comment:
do you want to allow 0 as a valid quantity?
like for example, i might run out of stock